src/components/job-overview/steps-list.js

```diff
@@ -36,15 +35,17 @@ export const StepsList = ({ appName, jobName, steps }) => {
       <Typography variant="h4">Steps</Typography>
       <div className="grid grid--gap-medium">
         {namedSteps.length > 0 ? (
-          namedSteps.map((step) => (
-            <div key={step.name} className="steps-list__step">
-              <div className="grid steps-list__divider">
-                <Icon className="step__icon" data={getStepIcon(step)} />
-                <span className="steps-list__divider-line"></span>
+          namedSteps
+            .sort((a, b) => (a.started < b.started ? -1 : 1))
```

Comment:
If two steps have the same value for "started", we should do a secondary sort/compare by "name" to make the sorted list more "stable".